### PR TITLE
Full 0.5/0.4 compat + fix failing tests

### PR DESCRIPTION
**Pull request by quinnj**

### CODE REVIEW

File: test/constructors.jl [hunk 1]
```diff
@@ -60,7 +60,7 @@ module TestConstructors
     @test isa(Z, NullableVector{Int})
     @test Z.isnull == [false, true, false, false, false, false, true]
 
-    Y = NullableArray([1, "na", 2, 3, 4, 5, "na"], Int, ASCIIString)
+    Y = NullableArray([1, "na", 2, 3, 4, 5, "na"], Int, Compat.ASCIIString)
```
Comment (nalimilan) [inline test/constructors.jl]: I'd rather change this into `Y = NullableArray([1, nothing, 2, 3, 4, 5, nothing], Int, Void)` or `Y = NullableArray(Nullable{Int}[1, Nullable(), 2, 3, 4, 5, Nullable()])`.

Comment (quinnj) [inline test/constructors.jl]: Updated

File: test/constructors.jl [hunk 2]
```diff
@@ -1,5 +1,5 @@
 module TestConstructors
-    using Base.Test
+    using Base.Test, Compat
```
Comment (nalimilan) [inline test/constructors.jl]: With the change, you don't need Compat AFAICT. Other than that, looks good to merge. Thanks for doing this!

**Comment by quinnj** [comment]
Updated

**Pull request merged**